validateAvroNestedSchema.py

```diff
@@ -56,24 +56,19 @@ def write_stamp_file(stamp_dict, output_dir):
     return out_path
 
 
-def write_avro_data(json, avro_schema):
-    """Encode json into avro format given a schema.
+def write_avro_data(json_data, json_schema):
+    """Encode json with fastavro module into avro format given a schema.
     """
-    writer = avro.io.DatumWriter(avro_schema)
     bytes_io = io.BytesIO()
-    encoder = avro.io.BinaryEncoder(bytes_io)
-    writer.write(json, encoder)
+    fastavro.schemaless_writer(bytes_io, json_schema, json_data)
     return bytes_io
 
 
-def read_avro_data(bytes_io, avro_schema):
-    """Read avro data and decode with a given schema.
+def read_avro_data(bytes_io, json_schema):
+    """Read avro data with fastavro module and decode with a given schema.
     """
-    raw_bytes = bytes_io.getvalue()
-    bytes_reader = io.BytesIO(raw_bytes)
-    decoder = avro.io.BinaryDecoder(bytes_reader)
-    reader = avro.io.DatumReader(avro_schema)
-    message = reader.read(decoder)
+    bytes_io.seek(0)
```

Comment:
Why do you need this `seek(0)`? Does that got to the start of the stream? Probably worth a note here.